sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -828,12 +831,17 @@ def test_warm_start_convergence():
     n_iter_cold_start = model.n_iter_
     assert n_iter_cold_start == n_iter_reference
 
-    # Fit the same model again, using a warm start: the optimizer just performs
-    # a single pass before checking that it has already converged
     model.set_params(warm_start=True)
     model.fit(X, y)
     n_iter_warm_start = model.n_iter_
-    assert n_iter_warm_start == 1
+    if sparse_X:
+        # TODO: sparse_enet_coordinate_descent is not yet updated.
```

Comment:
Similarly we should address this TODO before merging this PR.

Reply:
No for 2 reasons:
1. Let‘s keep this PR small
2. testing: If we only touch this one CD-function, our tests compare it to the others and garantees same results with high precision (small tol)